wheelsproxy/client.py

```diff
@@ -117,9 +121,24 @@ def get_package_releases(self, package_name):
                         .format(response.status_code, self.url, content))
             raise RuntimeError('Invalid response from index: {}'
                                .format(response.status_code))
-
-        releases = response.json()['releases']
-        return {k: self._clean_releases(v) for k, v in six.iteritems(releases)}
+        return response
+
+    def get_package_releases(self, package_name, ensure_serial=None):
+        for i in range(MAX_CACHE_BUSTING_RETRIES + 1):
+            response = self._request_package_releases(package_name)
+            if ensure_serial is not None:
+                serial = int(response.headers.get('X-PyPI-Last-Serial'))
```

Comment:
maybe this can fail more graceful if the header potentially changes in the future. 

we would not get notified about any change in advance and it would break the update mechanism. maybe try do update anyway like we do now without a sleep and log an error?

Reply:
Now that I look at it, we should instead fail if we reach MAX_CACHE_BUSTING_RETRIES.

We are notified via sentry + pagerduty of every new wheelsproxy exception... given that the update mechanism has been failing for the past couple of months, one or two days should not be a big deal... (besides, it changing would break lots more of the python ecosystem).